docs/python-api.rst

```diff
@@ -173,16 +173,19 @@ Tables
 ++++++
 
 The :ref:`tables API <sec_binary_interchange>` provides an efficient way of working
-with and interchanging :ref:`tree sequence data <sec_data_model>`. Each table
-class (e.g, :class:`NodeTable`, :class:`EdgeTable`) has a specific set of
-columns with fixed types, and a set of methods for setting and getting the data
+with and interchanging :ref:`tree sequence data <sec_data_model>`. Each table class
+(e.g, :class:`NodeTable`, :class:`EdgeTable`, :class:`SiteTable`) has a specific set
+of columns with fixed types, and a set of methods for setting and getting the data
 in these columns. The number of rows in the table ``t`` is given by ``len(t)``.
-Each table supports accessing the data either by row or column. To access the
-row ``j`` in table ``t`` simply use ``t[j]``. The value returned by such an
-access is an instance of :func:`collections.namedtuple`, and therefore supports
-either positional or named attribute access. To access the data in
-a column, we can use standard attribute access which will return a numpy array
-of the data. For example::
+Each table supports accessing the data either by row or column.
+
+To access the data in a *column*, we can use standard attribute access which will
+return a copy of the column data as a numpy array. To access the data in a *row*, say
+row number ``j`` in table ``t``, simply use ``t[j]``. The returned row has attributes
+allowing contents to be accessed by name, e.g. ``site_table[0].position``,
+``sites_table[0].ancestral_state``, ``sites_table[0].metadata`` etc. Note that row
+contents cannot be changed. To create a new row with changed content,
+row objects have a ``replace`` method. For example::
```

Comment:
Nice to link to a method, like "(e.g., :meth:`Site.replace`)" or something?

Reply:
Yeah, We don't list the actual method anywhere. I think either we'd need to list all the TableRow definitions, or perhaps just do:

```
.. autoclass:: Tskit.util.Dataclass
```

I did that, but the removed it, as then we'd be actually documenting this data class, and presumably therefore be committed to retaining it (and not e.g. implementing `replace` in a different way. So I thought it easiest not to commit to a particular implementation for the time being.

Reply:
OK, let's leave it then.